src/components/project/handlers/set-department-id.handler.ts

```diff
@@ -0,0 +1,85 @@
+import { ServerException } from '../../../common';
+import {
+  DatabaseService,
+  EventsHandler,
+  IEventHandler,
+  ILogger,
+  Logger,
+} from '../../../core';
+import { Project, ProjectStep } from '../dto';
+import { ProjectUpdatedEvent } from '../events';
+
+type SubscribedEvent = ProjectUpdatedEvent;
+
+@EventsHandler(ProjectUpdatedEvent)
+export class SetInitialEndDate implements IEventHandler<SubscribedEvent> {
+  constructor(
+    private readonly db: DatabaseService,
+    @Logger('project:set-department-id') private readonly logger: ILogger
+  ) {}
+
+  async handle(event: SubscribedEvent) {
+    this.logger.debug('Project mutation, set department id', {
+      ...event,
+      event: event.constructor.name,
```

Comment:
This was really only on that one handler because it was handling so many different types of events. It will just be a constant on this one though so it doesn't provide relevant info.